drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -1066,6 +1068,38 @@ def _is_answer_correct(self):
         """
         return self._answer_correctness() == self.SOLUTION_CORRECT
 
+    def calculate_due_date(self):
+        """
+        Return the date submissions should be closed from.
+
+        Calculates and return the due date after which the submissions
+        will no longer be accepted. Returns None if both due & graceperiod
+        are missing which happens for pure DragAndDropXblock as they are
+        not its attributes. They are present for DragAndDropXblockWithMixins
```

Comment:
Please say specifically that the `due` and `graceperiod` fields get added by [`InheritanceMixin`](https://github.com/edx/edx-platform/blob/7eede0926787f3d991916e28da2b4651845e472e/common/lib/xmodule/xmodule/modulestore/inheritance.py#L31-L74).